File: TaskManager/MyTaskManager/Command.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MyTaskManager
{
    public abstract class Command
    {
        public string Description { get; set; }
        public DateTime Timestamp { get; set; }
        public CommandType Type { get; set; }

        public Command()
        {
            Timestamp = DateTime.Now;
        }

        public abstract void Execute();
        public abstract void Undo();
    }
}

File: TaskManager/TaskManagerSolution/src/Command.java
import java.time.LocalDateTime;

public abstract class Command {
    private String description;
    private LocalDateTime timestamp;
    private CommandType type;

    public Command(String description, CommandType type) {
        this.description = description;
        this.timestamp = LocalDateTime.now();
        this.type = type;
    }

    public abstract void execute();
    public abstract void undo();

    public String getDescription() {
        return description;
    }

    public LocalDateTime getTimestamp() {
        return timestamp;
    }

    public CommandType getType() {
        return type;
    }

}
